atproto/identity/cache_catalog.go

```diff
@@ -0,0 +1,209 @@
+package identity
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/bluesky-social/indigo/atproto/syntax"
+)
+
+type CacheCatalog struct {
+	HitTTL        time.Duration
+	ErrTTL        time.Duration
+	Inner         Catalog
+	mutex         sync.RWMutex
+	handleCache   map[syntax.Handle]HandleEntry
+	identityCache map[syntax.DID]IdentityEntry
+}
```

Comment:
Do we wanna use https://github.com/hashicorp/golang-lru for these caches to limit their sizes and help with eviction?

Reply:
I was thinking about that. I kind of want to keep this package more minimal w/r/t deps and have sub-packages or something for fancier implementations. But an LRU is pretty simple/common and we should probably have a polished and really robust solution to caching, as it is pretty load-bearing for identity.